### PR TITLE
Use system proxy settings with atlassian httpClient

### LINKED ISSUE #16: Plugin ignores noProxyHosts
After quite some search we found a problem with the noProxyHosts list. We pass Bitbucket a list of noProxyHosts (in form of "*.company-name.com" which are internally used domains) and it can connect to our Jira instance.

When using the plugin, it is running into a timeout when trying to connect to our Jenkins. Deactivating this list results in the plugin being able to connect to the Jenkins, but Bitbucket needs this Proxy otherwise it cannot connect to the Marketspace (or anything else outside our internal network).

### PR DESCRIPTION
**Pull request by KamikaZeeFu**
Fixes #16 

Instead of using a httpClient wired in through OSGI, create a custom one that will respect system properties, especially regarding proxy settings.

This PR also contains some minor clean ups. Especially the removal of package exports was necessary to get this functionality to work properly.

Tests for this feature were not feasible, due to the complexity of OSGI.

### CODE REVIEW

File: src/main/resources/META-INF/spring/atlassian-plugins-component-imports.xml [hunk 1]
```diff
@@ -1,17 +0,0 @@
-<beans:beans xmlns:beans="http://www.springframework.org/schema/beans"
```
Comment (Eernie) [inline src/main/resources/META-INF/spring/atlassian-plugins-component-imports.xml]: Hmm, whats the reason that this file is removed? How will Spring now determine the injectables?

**Comment by Eernie** [comment]
Looks ok, Just need to know about the `atlassian-plugins-component-imports.xml`.
Thanks for giving me the option for the type of http client :+1:

**Comment by KamikaZeeFu** [comment]
While trying to figure out how atlassian plugins work, I read the documentation on the atlassian spring scanner plugin. 

Basically, it looks for `@ComponentImport` and a few other Annotations an generates the required information that way. That's why this file wasn't needed, since  the annotations are already used everywhere. 


See 
https://bitbucket.org/atlassian/atlassian-spring-scanner/src/dd6c0b0fac5d7c7a571c48d6ff9f97e224df8dff?at=atlassian-spring-scanner-parent-1.2.9 for the documentation source. 

I hope this clarifies the change.

**Comment by Eernie** [comment]
Thats great, thanks for that. Can you get it ready for merge? :+1:

**Comment by KamikaZeeFu** [comment]
PR is rebased against current master.

**Pull request merged**